lighthouse-core/lib/network-request.js

```diff
@@ -341,10 +364,99 @@ module.exports = class NetworkRequest {
+
+    //    StartTime
+    //    | ConnectStart
+    //    | |     SSLStart  SSLEnd
+    //    | |     |         | ConnectEnd
+    //    | |     |         | | SendStart/End   ReceiveHeadersEnd
+    //    | |     |         | | |               |                EndTime
+    //    ▼ ▼     ▼         ▼ ▼ ▼               ▼                ▼
+    //    [ [TCP  [   SSL   ] ] [   Request   ] [   Response   ] ]
+    //    ▲ ▲     ▲         ▲ ▲ ▲             ▲ ▲              ▲ ▲
+    //    | |     '-SSLMs---' | '-requestMs---' '-responseMs---' |
+    //    | '----TCPMs--------'                                  |
+    //    |                                                      |
+    //    '------------------------TotalMs-----------------------'
+
+    const totalHeader = this.responseHeaders.find(item => item.name === HEADER_TOTAL);
+    // Bail if there was no totalTime.
+    if (!totalHeader) return;
+
+    const totalMs = parseInt(totalHeader.value);
+
+    const TCPMsHeader = this.responseHeaders.find(item => item.name === HEADER_TCP);
+    const requestMsHeader = this.responseHeaders.find(item => item.name === HEADER_REQ);
+    const SSLMsHeader = this.responseHeaders.find(item => item.name === HEADER_SSL);
+    const responseMsHeader = this.responseHeaders.find(item => item.name === HEADER_RES);
+
+    // TODO(exterkamp): NaN checking
+    // Make sure all Times are initialized and are non-negative.
+    const TCPMs = TCPMsHeader ? Math.max(0, parseInt(TCPMsHeader.value)) : 0;
+    const requestMs = requestMsHeader ? Math.max(0, parseInt(requestMsHeader.value)) : 0;
+    const _/** SSLMs */ = SSLMsHeader ? Math.max(0, parseInt(SSLMsHeader.value)) : 0;
+    const responseMs = responseMsHeader ? Math.max(0, parseInt(responseMsHeader.value)) : 0;
+
+    // Bail if the timings don't add up.
+    if (TCPMs + requestMs + responseMs !== totalMs) {
+      return;
+    }
+
+    /*
+    This code will change performance scores.  The current Lighthouse version is 4.x,
+    and does not take into account these changes.  In order to uncomment and apply these
+    changes the main version number will need to be bumped to 5.x+.
+
+    See: https://github.com/GoogleChrome/lighthouse/issues/7752
+
+    const origEnd = this.endTime;
+
+    // EndTime and responseReceivedTime are in seconds, so conversion is necessary.
+    this.endTime = this.startTime + (totalMs / 1000);
+    this.responseReceivedTime = this.startTime + ((TCPMs + requestMs) / 1000);
+
+    this.timing.connectStart = 0;
+    this.timing.connectEnd = TCPMs;
+    // Make sure that SSL time is less than or equal to total TCP time.
+    if (SSLMs <= TCPMs) {
+      this.timing.sslStart = TCPMs - SSLMs;
+      this.timing.sslEnd = TCPMs;
+    }
+    this.timing.sendStart = TCPMs;
+    this.timing.sendEnd = TCPMs;
+    this.timing.receiveHeadersEnd = TCPMs + requestMs;
+    */
+
+    this.endTimeDeltaMs = (this.endTime - (this.startTime + (totalMs / 1000))) * 1000;
```

Comment:
oh, I think this actually should be nested under `lrStatistics` on the network request object, just for organizational purposes

Reply:
Done.